src/main/java/no/nav/familie/integrasjoner/dokdist/DokdistController.kt

```diff
@@ -0,0 +1,30 @@
+package no.nav.familie.integrasjoner.dokdist
+
+import no.nav.familie.kontrakter.felles.Ressurs
+import no.nav.familie.kontrakter.felles.Ressurs.Companion.success
+import no.nav.security.token.support.core.api.ProtectedWithClaims
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@ProtectedWithClaims(issuer = "azuread")
+@RequestMapping("/api/dist")
+class DokdistController(private val dokdistService: DokdistService) {
+
+
+
+    @GetMapping("v1/{journalpostId}")
+    @ResponseStatus(HttpStatus.OK)
+    fun distribuerJournalpost(@PathVariable(name = "journalpostId") journalpostId: String): ResponseEntity<Ressurs<String>>  {
+
+        val resp = dokdistService.distribuerDokumentForJournalpost(journalpostId)
```

Comment:
Kan vi wrappe servicekallet i en runCatching? På den måten håndterer vi automatisk alle feil som skjer i servicen og kan logge en sane melding vi kjenner.

Reply:
```suggestion
        return Result.runCatchin { dokdistService.distribuerDokumentForJournalpost(journalpostId) }
            .onSuccess { ResponseEntity.ok(success(it.bestillingsId)) }
            .onFailure { ResponseEntity.failure(failure("Feilmelding", it)) } // Velg riktig responsstatus, husker ikke i farta hva den heter.
```

Servicekallet bør ikke kunne returnere null, isåfall kan du wrappe det i en when i tillegg for å håndtere det i controlleren, samt skrive tester for det.

Reply:
Det kan jeg fikse, men trenger vi å returnere noe hvis det feiler? Er ikke det mest når vi skal videreformidle noe til front-end?

Reply:
Nei, trenger ikke returnere noe spesielt tenker jeg. Bare videreføre feilmeldingen som kommer fra tjenesten vi kaller slik at ba-sak tilslutt har den :)